sklearn/utils/_estimator_html_repr.py

```diff
@@ -424,22 +369,128 @@ def estimator_html_repr(estimator):
             " HTML representation is unable to render, please try loading this page"
             " with nbviewer.org."
         )
-        out.write(
+        html_template = (
             f"<style>{style_with_id}</style>"
             f'<div id="{container_id}" class="sk-top-container">'
             '<div class="sk-text-repr-fallback">'
             f"<pre>{html.escape(estimator_str)}</pre><b>{fallback_msg}</b>"
             "</div>"
             '<div class="sk-container" hidden>'
         )
+
+        out.write(html_template)
+
         _write_estimator_html(
             out,
             estimator,
             estimator.__class__.__name__,
             estimator_str,
             first_call=True,
+            is_fitted_css_class=is_fitted_css_class,
+            is_fitted_icon=is_fitted_icon,
         )
         out.write("</div></div>")
 
         html_output = out.getvalue()
         return html_output
+
+
+class _HTMLDocumentationLinkMixin:
+    """Mixin class allowing to generate a link to the API documentation.
+
+    This mixin relies on three attributes:
+    - `_doc_link_module`: it corresponds to the root module (e.g. `sklearn`). Using this
+      mixin, the default value is `sklearn`.
+    - `_doc_link_template`: it corresponds to the template used to generate the
+      link to the API documentation. Using this mixin, the default value is
+      `"https://scikit-learn.org/{version_url}/modules/generated/
+      {estimator_module}.{estimator_name}.html"`.
+    - `_url_param_generator`: it corresponds to a function that generates the parameters
+      to be used in the template when the estimator module and name are not sufficient.
+
+    The method :meth:`_get_doc_link` generates the link to the API documentation for a
+    given estimator.
+
+    This useful provides all the necessary states for
+    :func:`sklearn.utils.estimator_html_repr` to generate a link to the API
+    documentation for the estimator HTML diagram.
+
+    Examples
+    --------
+    If the default values for `_doc_link_module`, `_doc_link_template` are not suitable,
+    then you can override them:
+    >>> from sklearn.base import BaseEstimator
+    >>> estimator = BaseEstimator()
+    >>> estimator._doc_link_template = "https://website.com/{single_param}.html"
+    >>> def url_param_generator(estimator):
+    ...     return {"single_param": estimator.__class__.__name__}
+    >>> estimator._url_param_generator = url_param_generator
+    >>> estimator._get_doc_link()
+    'https://website.com/BaseEstimator.html'
+    """
+
+    @property
+    def _doc_link_module(self):
+        return getattr(self, "__doc_link_module", "sklearn")
+
+    @_doc_link_module.setter
+    def _doc_link_module(self, value):
+        setattr(self, "__doc_link_module", value)
```

Comment:
Can you explain why it makes sense to have this indirection via a getter and a setter? Especially because we don't do any validation/modification during the setting. (same comment applies to the other properties)